src/LmdbCache.ts

```diff
@@ -75,6 +76,15 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
     await this.db.put(`${stateCacheKey.contractTxId}|${stateCacheKey.sortKey}`, value);
   }
 
+  async delete(contractTxId: string): Promise<void> {
+    await this.db.transaction(() => {
+      this.db.getKeys({ start: `${contractTxId}|${lastPossibleKey}`, reverse: true })
```

Comment:
hmm...isn't there a risk that with the current implementation we will remove the contract that is "before" the `contractTxId`?

e.g. we have keys ('A' and 'B' is contractTxId)
A|1, A|2, A|3, B|1, B|2

and we decide to remove the contract 'B'.

Because we're starting with last possible key for the contract 'B' and moving in reverse order (without checking if we haven't moved "out" of the contractTxId that we want to remove) - wouldn't we also remove the contract 'A' in this case?

The safest solution IMHO would be:
this.db.getKeys({ start: `${contractTxId}|${genesisSortKey}`, end: `${contractTxId}|${lastPossibleKey}`` })

Reply:
You are completely right, I added a better test that removes from the middle of the history